File: Python-Library-Project/film.py
#Sumeet Tulsani
from media import Media

class Film(Media):
    def __init__(self,call_number,title,subjects,director,notes,year):
        Media.__init__(self,call_number,title,subjects,notes)
        self.director = director
        self.year = year

    def display(self):
        print ("---------------FILM-----------------");
        print ("CALL NUMBER::",self.call_number);
        print ("TITLE::",self.title);
        print ("SUBJECTS::",self.subjects);
        print ("DIRECTOR::",self.director);
        print ("NOTES::",self.notes);
        print ("YEAR::",self.year);

    def compare_other(self,searchby):
        if (self.director.find(searchby) != -1) or  (self.notes.find(searchby) != -1) or (self.year.find(searchby) != -1):
            return True
        else:
            return False

File: LibraryProject/film.cpp
#include"film.h"
#include<iostream>
#include<string>
#include<vector>
#include "boost/algorithm/string.hpp"


using namespace std;
using namespace boost;

Film::Film(const string& call_number,const string& title,const string& subjects,
           const string& director,const string& notes,const string& year):
           Media(call_number,title,subjects,notes),director(director),year(year)
{

}

void Film::display() const
{
    cout << "-----------FILM----------" << endl;
    cout << "CALL NUMBER::" << call_number << endl << "TITLE::" << title << endl << "SUBJECTS::"
     << subjects << endl << "DIRECTOR::" << director << endl << "NOTES::" <<notes << endl <<
      "YEAR::" << year << endl;
}

bool Film::compare_other(const string& searchby) const
{
    size_t found = notes.find(searchby);
    size_t found1 = director.find(searchby);
    size_t found2 = year.find(searchby);
    if((found!=string::npos) || (found1!=string::npos) || (found2!=string::npos))
    {
        return true;
    }
    else
    {
        return false;
    }
}

